File: market/integration-test-extra-3/java/ru/yandex/market/fulfillment/stockstorage/SupportControllerTest.java
package ru.yandex.market.fulfillment.stockstorage;

import com.github.springtestdbunit.annotation.DatabaseSetup;
import com.github.springtestdbunit.annotation.ExpectedDatabase;
import com.github.springtestdbunit.assertion.DatabaseAssertionMode;
import org.junit.jupiter.api.Test;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;


@DatabaseSetup("classpath:database/states/support_controller/setup.xml")
public class SupportControllerTest extends AbstractContextualTest {

    @ExpectedDatabase(
            value = "classpath:database/expected/support_controller/park.xml",
            assertionMode = DatabaseAssertionMode.NON_STRICT_UNORDERED)
    @Test
    public void park() throws Exception {
        mockMvc.perform(post("/support/execution-queue/park")
                .content("{\"ids\" : [1, 2, 3, 4], \"days\" : 2}")
                .contentType("application/json"))
                .andExpect(status().isOk())
                .andReturn().getResponse()
                .getContentAsString();
    }

    @ExpectedDatabase(
            value = "classpath:database/expected/support_controller/delete.xml",
            assertionMode = DatabaseAssertionMode.NON_STRICT_UNORDERED)
    @Test
    public void delete() throws Exception {
        mockMvc.perform(
                post("/support/execution-queue/delete")
                        .content("{\"ids\":[2, 3, 4]}")
                        .contentType("application/json"))
                .andExpect(status().isOk())
                .andReturn().getResponse()
                .getContentAsString();
    }

    @Test
    @ExpectedDatabase(
            value = "classpath:database/states/support_controller/setup.xml",
            assertionMode = DatabaseAssertionMode.NON_STRICT_UNORDERED)
    public void deleteFail() throws Exception {
        mockMvc.perform(
                post("/support/execution-queue/delete")
                        .content("{\"ids\":[1, 2, 3, 4]}")
                        .contentType("application/json"))
                .andExpect(status().isBadRequest())
                .andReturn().getResponse()
                .getContentAsString();
    }

}

File: market/test/kotlin/ru/yandex/market/wms/core/controller/SupportControllerTest.kt
package ru.yandex.market.wms.core.controller

import com.github.springtestdbunit.annotation.DatabaseSetup
import com.github.springtestdbunit.annotation.ExpectedDatabase
import com.github.springtestdbunit.assertion.DatabaseAssertionMode
import org.junit.jupiter.api.Test
import org.springframework.http.MediaType
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
import org.springframework.test.web.servlet.result.MockMvcResultMatchers
import ru.yandex.market.wms.common.spring.IntegrationTest
import ru.yandex.market.wms.common.spring.utils.FileContentUtils

class SupportControllerTest : IntegrationTest() {

    @Test
    @DatabaseSetup("/controller/support/write-off/before.xml")
    @ExpectedDatabase("/controller/support/write-off/after.xml", assertionMode = DatabaseAssertionMode.NON_STRICT_UNORDERED)
    fun writeOffSomeSerialNumbers() {
        mockMvc.perform(MockMvcRequestBuilders.post("/support/write-off")
                .contentType(MediaType.APPLICATION_JSON)
                .content(FileContentUtils.getFileContent("controller/support/write-off/request.json")))
                .andExpect(MockMvcResultMatchers.status().is2xxSuccessful)
    }
}
